### PR TITLE
Implement grammar info

### LINKED ISSUE #29: Implement GrammarInfo with GrammarGender and GrammarNumber

### PR DESCRIPTION
**Pull request by HassanAkbar**
- Added `Gender` and `Name` to GrammarInfo
- Extracted common functionality to the `Enum` module

Fixes #29

### CODE REVIEW

File: lib/glossarist/designation/grammar_info.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+      def to_h
+        {
+          "preposition" => preposition,
+          "participle" => participle,
+          "adjective" => adjective,
+          "verb" => verb,
+          "adverb" => adverb,
+          "noun" => noun,
+          "gender" => gender.join(", "),
```
Comment (ronaldtse) [inline lib/glossarist/designation/grammar_info.rb:58]: Returning an array would be better so we can preserve the structure.

**Review by ronaldtse — CHANGES_REQUESTED** [review]
Can we add a few real examples from iev-data?

**Comment by HassanAkbar** [comment]
> Can we add a few real examples from iev-data?

I've added the real payload from iev-data in [this](https://github.com/glossarist/glossarist-ruby/pull/31/files#diff-38f3c78cfdf3eae0598a5be293cd595ccfff28c6f91251ca6855f55501f1adfaR171) test case.

File: lib/glossarist/designation/expression.rb [hunk 2]
```diff
@@ -5,15 +5,38 @@
...
       # List of grammar info.
       # @return [Array<GrammarInfo>]
-      attr_accessor :grammar_info
+      attr_reader :grammar_info
+
+      def grammar_info=(grammar_info)
+        @grammar_info ||= []
+
+        grammar_info.each do |info|
+          @grammar_info << GrammarInfo.new(info)
```
Comment (ronaldtse) [inline lib/glossarist/designation/expression.rb:19]: What happens in:
```ruby
x = Expression.new
x.grammar_info = [x,y,z] # assuming x y z are things GrammarInfo consumes
x.grammar_info = [x]
```
?

**Review by ronaldtse — CHANGES_REQUESTED** [review]
Just one comment, thanks!

File: lib/glossarist/designation/expression.rb (hunk 2 above)
Comment (HassanAkbar) [reply lib/glossarist/designation/expression.rb:19]: Good catch. 
I have updated the PR.

Comment (ronaldtse) [reply lib/glossarist/designation/expression.rb:19]: @HassanAkbar thanks. In this case do you just want to use `map`? i.e.
```ruby
grammar_info.map do |g|
  GrammarInfo.new(g)
end
```

Comment (HassanAkbar) [reply lib/glossarist/designation/expression.rb]: Yes, this is much better. Thank you.
I’ve updated the code.

**Comment by ronaldtse** [comment]
@HassanAkbar can you please rebase and merge? Thanks.

**Review by ronaldtse — APPROVED** [review]

**Pull request merged**